core/scripts/vrfv2/superscripts/main.go

```diff
@@ -0,0 +1,257 @@
+
+		coordinator, err := vrf_coordinator_v2.NewVRFCoordinatorV2(coordinatorAddress, ec)
+		helpers.PanicErr(err)
+
+		fmt.Println("\nSetting Config...")
+		tx, err = coordinator.SetConfig(owner,
+			uint16(3),     // minRequestConfirmations
+			uint32(2.5e6), // max gas limit
+			uint32(86400), // stalenessSeconds
+			uint32(33285), // gasAfterPaymentCalculation
+			decimal.RequireFromString(*fallbackWeiPerUnitLink).BigInt(), // 0.01 eth per link fallbackLinkPrice
+			vrf_coordinator_v2.VRFCoordinatorV2FeeConfig{
+				FulfillmentFlatFeeLinkPPMTier1: uint32(500),
+				FulfillmentFlatFeeLinkPPMTier2: uint32(500),
+				FulfillmentFlatFeeLinkPPMTier3: uint32(500),
+				FulfillmentFlatFeeLinkPPMTier4: uint32(500),
+				FulfillmentFlatFeeLinkPPMTier5: uint32(500),
+				ReqsForTier2:                   big.NewInt(0),
+				ReqsForTier3:                   big.NewInt(0),
+				ReqsForTier4:                   big.NewInt(0),
+				ReqsForTier5:                   big.NewInt(0),
+			},
+		)
+		helpers.PanicErr(err)
+		confirmTXMined(context.Background(), ec, tx, chainID)
+
+		fmt.Println("\nConfig set, getting current config from deployed contract...")
+		cfg, err := coordinator.GetConfig(nil)
+		helpers.PanicErr(err)
+		feeConfig, err := coordinator.GetFeeConfig(nil)
+		helpers.PanicErr(err)
+		fmt.Printf("Config: %+v\n", cfg)
+		fmt.Printf("Fee config: %+v\n", feeConfig)
+
+		fmt.Println("\nRegistering proving key...")
+		if strings.HasPrefix(*registerKeyUncompressedPubKey, "0x") {
+			*registerKeyUncompressedPubKey = strings.Replace(*registerKeyUncompressedPubKey, "0x", "04", 1)
+		}
+		pubBytes, err := hex.DecodeString(*registerKeyUncompressedPubKey)
+		helpers.PanicErr(err)
+		pk, err := crypto.UnmarshalPubkey(pubBytes)
+		helpers.PanicErr(err)
+		tx, err = coordinator.RegisterProvingKey(owner,
+			common.HexToAddress(*registerKeyOracleAddress),
+			[2]*big.Int{pk.X, pk.Y})
+		helpers.PanicErr(err)
+		confirmTXMined(
+			context.Background(),
+			ec,
+			tx,
+			chainID,
+			fmt.Sprintf("Uncompressed public key: %s,", *registerKeyUncompressedPubKey),
+			fmt.Sprintf("Oracle address: %s,", *registerKeyOracleAddress),
+		)
+
+		fmt.Println("\nProving key registered, getting proving key hashes from deployed contract...")
+		_, _, s_provingKeyHashes, err := coordinator.GetRequestConfig(nil)
+		helpers.PanicErr(err)
+		fmt.Printf("Hashes: %+v\n", s_provingKeyHashes)
+
+		// Deploy consumer and subscribe.
+		fmt.Println("\nDeploying consumer and subscribing...")
+		keyHashBytes := common.HexToHash(*keyHash)
+		_, tx, _, err = vrf_single_consumer_example.DeployVRFSingleConsumerExample(
+			owner,
+			ec,
+			coordinatorAddress,
+			common.HexToAddress(*linkAddress),
+			uint32(1000000), // gas callback
+			uint16(5),       // confs
+			uint32(1),       // words
+			keyHashBytes)
+		helpers.PanicErr(err)
+		consumerAddress := confirmContractDeployed(context.Background(), ec, tx, chainID)
+		subId := uint64(1)
```

Comment:
Doesn't this script need to create this sub so that it exists before we fund it?

Reply:
Also should add the consumer as a consumer of the subscription

Reply:
https://github.com/smartcontractkit/chainlink/blob/217aa407349bf2fe91f2edfcbbaffee8960f75e9/contracts/src/v0.8/tests/VRFSingleConsumerExample.sol#L43

The single consumer contract creates the subscription and adds itself upon from its own constructor, which I didn't expect initially.

Reply:
Ahh I didn't realize you were using this instead of VRFExternalSubOwnerExample. I'd recommend using VRFExternalSubOwnerExample, SingleConsumerExample doesn't let you change request configs like requestConfirmations, keyHash etc after you deploy it which makes it a little less useful for testing.